app/models/host/base.rb

```diff
@@ -192,6 +192,7 @@ def populate_fields_from_facts(parser, type, source_proxy)
       set_non_empty_values(parser, attributes_to_import_from_facts)
       set_interfaces(parser) if parser.parse_interfaces?
       set_reported_data(parser)
+      set_comment(parser)
```

Comment:
How about:
```suggestion
      set_comment(parser) if parser.comment?
```

Then the implementation would be (untested):
```ruby
def comment?
  facts.has_key(:foreman_comment)
end
```

This would allow the user to clear comments using the empty string but also safely detect whether the fact actually exists.